x-pack/solutions/security/plugins/security_solution/docs/testing/test_plans/detection_response/prebuilt_rules/prebuilt_rule_import.md

```diff
@@ -91,195 +119,429 @@ User stories:
+```
+
+#### **Scenario: Converting a prebuilt rule to a custom rule on import**
+
+This is an edge case that can happen when the user manually edits an ndjson file and:
+
+- By mistake, makes a legit custom rule in it prebuilt by updating the `immutable` and `rule_source` fields.
+
+**Automation**: 1 API integration test.
+
+```Gherkin
+Given the import payload contains a prebuilt rule
 And its rule_id does NOT match any rule assets from the installed package
-And this custom rule has already been created
+And this rule is not installed
 When the user imports the rule
-Then the rule should be updated
-And the ruleSource type should be "internal"
-And the "version" field should be set to the existing rule's "version"
+Then the rule should be created
+And the created rule should be custom
+And the created rule's parameters should match the import payload
 ```
 
-#### Scenario: Importing a new custom rule missing a `version` field
+#### **Scenario: Making an imported custom rule upgradeable to a prebuilt rule**
+
+This is an edge case that can happen when Elastic ships a new prebuilt rule with a `rule_id` accidentally matching the `rule_id` of a user's custom rule.
 
-**Automation**: 1 integration test.
+**Automation**: 1 API integration test.
 
 ```Gherkin
-Given the import payload contains a custom rule without a version field
+Given the import payload contains a custom rule
 And its rule_id does NOT match any rule assets from the installed package
-And this custom rule hasn't been created yet
+And this rule is not installed
+When the user imports the rule
+Then the rule should be created
+And the created rule should be custom
+When a new version of the package gets installed
+And a new prebuilt rule asset is added in it with the same rule_id and a higher version
+Then the rule should be upgradeable to the asset's version as if it were a prebuilt rule
+When the user upgrades this rule to the TARGET version
+Then the rule should be upgraded to the asset's version
+And the upgraded rule should be prebuilt
+And the upgraded rule should be marked as non-customized
+And the upgraded rule's parameters should match the asset
+```
+
+### Handling missing base versions
+
+NOTE: These are not edge cases but rather normal cases. In many package upgrade scenarios, hundreds of prebuilt rules (out of ~1300 of them at the time of writing) won't have a base version.
+
+**When the base version** of a prebuilt rule that is being imported **is missing** among the `security-rule` asset saved objects, and the user imports this rule, the following scenarios apply.
+
+#### **Scenario: Importing a prebuilt rule with a missing base version when it's not installed**
+
+If this rule is not installed, it should be created with `is_customized` field set to `false`.
+
+**Automation**: 1 API integration test.
+
+```Gherkin
+Given the import payload contains a prebuilt rule
+And its rule_id matches one or a few rule assets from the installed package
```

Comment:
```suggestion
And its rule_id matches one or more prebuilt rule assets from the installed package
```

Reply:
@maximpn In these test plans we just say "rule asset" without adding "prebuilt" - it is assumed. Since there are no custom rule assets, we can afford being more succinct. The term "rule asset" is explained in the common terminology.

Reply:
`one or more` 👍